controllers/kubeobjects.go

```diff
@@ -47,84 +49,111 @@ func kubeObjectsRecover(
-		"source namespace", sourceNamespaceName,
+		"source namespace", sourceNamespacedName.Namespace,
 		"target namespace", targetNamespaceName,
 		"capture number", captureNumber,
 	)
 
-	namespacedName, err := namespacedName(veleroNamespaceName, s3Url, s3BucketName, captureNumber)
-	if err != nil {
-		return err
-	}
+	// generate backup name
+	restoreNamespacedName := getBackupNamespacedName(sourceNamespacedName.Name,
+		sourceNamespacedName.Namespace, fmt.Sprintf("%d", index), captureNumber)
+
+	backupNamespacedName := getBackupNamespacedName(sourceNamespacedName.Name,
+		sourceNamespacedName.Namespace, restoreSpec.BackupName, captureNumber)
 
 	return backupDummyCreateAndIfAlreadyExistsThenRestore(
-		namespacedName,
+		restoreNamespacedName,
+		backupNamespacedName,
 		objectWriter{ctx: ctx, Writer: writer, log: log},
 		reader,
 		s3Url,
 		s3BucketName,
 		s3KeyPrefix,
-		sourceNamespaceName,
+		sourceNamespacedName.Namespace,
 		targetNamespaceName,
+		restoreSpec,
+		captureNumber,
 	)
 }
 
 func backupDummyCreateAndIfAlreadyExistsThenRestore(
-	namespacedName types.NamespacedName,
+	restoreNamespacedName types.NamespacedName,
+	backupNamespacedName types.NamespacedName,
 	w objectWriter,
 	reader client.Reader,
 	s3Url string,
 	s3BucketName string,
 	s3KeyPrefix string,
 	sourceNamespaceName string,
 	targetNamespaceName string,
+	restoreSpec ramendrv1alpha1.ResourceRestoreInstance,
+	captureNumber int64,
 ) error {
 	backupLocation, backup, err := w.backupCreate(
-		namespacedName, s3Url, s3BucketName, s3KeyPrefix, backupSpecDummy(), sourceNamespaceName,
+		backupNamespacedName, s3Url, s3BucketName, s3KeyPrefix, backupSpecDummy(), sourceNamespaceName, captureNumber,
 	)
 	if err != nil {
 		return err
 	}
 
-	if err := reader.Get(w.ctx, namespacedName, backup); err != nil {
+	backupObjectNamespacedName := types.NamespacedName{
+		Name:      backupNamespacedName.Name,
+		Namespace: VeleroNamespaceNameDefault,
+	}
+	if err := reader.Get(w.ctx, backupObjectNamespacedName, backup); err != nil {
 		return pkgerrors.Wrap(err, "dummy backup get")
 	}
 
-	return backupDummyStatusProcess(backupLocation, backup, namespacedName, w, reader,
+	// checks status of dummy backup and possibly starts restore
+	return backupDummyStatusProcess(backupLocation, backup, backupNamespacedName,
+		restoreNamespacedName,
+		w, reader,
 		sourceNamespaceName,
 		targetNamespaceName,
+		restoreSpec,
 	)
 }
 
 func backupDummyStatusProcess(
 	backupLocation *velero.BackupStorageLocation,
 	backup *velero.Backup,
-	namespacedName types.NamespacedName,
+	backupNamespacedName types.NamespacedName,
+	restoreNamespacedName types.NamespacedName,
 	w objectWriter,
 	reader client.Reader,
 	sourceNamespaceName string,
 	targetNamespaceName string,
+	restoreSpec ramendrv1alpha1.ResourceRestoreInstance,
 ) error {
 	backupStatusLog(backup, w.log)
 
+	w.log.Info(fmt.Sprintf("backupDummyStatusProcess: %s Status.Phase=%s", backup.Name, backup.Status.Phase))
+
```

Comment:
(nit) The goal of `backupStatusLog` is to log all backup status.  Looks like it does lack backup name, so that could be added, it has phase.  It could take an string to indicate "dummy" or "real"